lib/types.ts

```diff
@@ -251,7 +251,7 @@ export type DeepOmit<T extends DeepModify<Filter>, Filter> = T extends Builtin
       { [K in Exclude<RequiredKeys<T>, keyof Filter>]: T[K] } &
       OmitProperties<
         {
-          [K in Extract<RequiredKeys<T>, keyof Filter>]: Filter[K] extends true
+          [K in Extract<RequiredKeys<T>, keyof Filter>]: Filter[K] extends never
```

Comment:
Given `T extends never` we distribute the type, but not checking that it is `never` (see [Distributive Conditional Types](https://www.typescriptlang.org/docs/handbook/advanced-types.html#distributive-conditional-types))

To check that the type is `never` you need to use this types specifically:

```ts
type IsNever<T> = [T] extends [never] ? true : false;
```